### PR TITLE
fix 4420(AirsimROSWrapper::airsim_timestamp_to_ros failed to compile in Visual Studio)

### LINKED ISSUE #4420: AirsimROSWrapper::airsim_timestamp_to_ros failed to compile in Visual Studio
I ran airsim_node of airsim_ros2 package in Windows 10. There is a key issue to report.

Cannot compile 
https://github.com/microsoft/AirSim/blob/77a879e8b6459543caa1ab867e277f4958becb91/ros2/src/airsim_ros_pkgs/src/airsim_ros_wrapper.cpp#L912

**Errors:**

Severity	Code	Description	Project	File	Line	Suppression State

Error	C2664	'std::chrono::time_point<std::chrono::system_clock,std::chrono::duration<std::chrono::system_clock::rep,std::chrono::system_clock::period>>::time_point(const _Duration &) noexcept': cannot convert argument 1 from 'std::chrono::nanoseconds' to 'const _Duration &'	ros2airsim	D:\app\ros2ex\airsim_ros\airsim_ros_pkgs\src\airsim_ros_wrapper.cpp	924	

**Solution:**
For converting airsim timestamp to rclcpp::Time, I think the following codes can reach the goal
rclcpp::Time AirsimROSWrapper::airsim_timestamp_to_ros(const msr::airlib::TTimePoint& stamp) const
{
    // airsim appears to use chrono::system_clock with nanosecond precision
    //std::chrono::nanoseconds dur(stamp);
    //std::chrono::time_point<std::chrono::system_clock> tp(dur);
    //rclcpp::Time cur_time = chrono_timestamp_to_ros(tp);
    return rclcpp::Time(stamp);
}
instead of
rclcpp::Time AirsimROSWrapper::airsim_timestamp_to_ros(const msr::airlib::TTimePoint& stamp) const
{
    // airsim appears to use chrono::system_clock with nanosecond precision
    std::chrono::nanoseconds dur(stamp);
    std::chrono::time_point<std::chrono::system_clock> tp(dur);
    rclcpp::Time cur_time = chrono_timestamp_to_ros(tp);
    return cur_time;
}

### PR DESCRIPTION
**Pull request by dzywater**
Fixes: https://github.com/microsoft/AirSim/issues/4420

### CODE REVIEW

**Comment by zimmy87** [comment]
Hey @dzywater, thanks for submitting this PR! I'm trying to test this on my end, but I can't build the ros2 wrapper on Windows due to a missing package ([tf2_geometry_msgs](https://index.ros.org/p/tf2_geometry_msgs/github-ros2-geometry2/#galactic)). Do you know how to install ROS packages on Windows? The only guide I could find is [here](https://ms-iot.github.io/ROSOnWindows/GettingStarted/UsingROSonWindows.html), but I can't find the package on Chocolatey and rosdep doesn't appear to be working for me.

**Comment by dzywater** [comment]
Did you compile ros2 yourself on Windows?
I downloaded ros2foxy precomiled binary on https://github.com/ros2/ros2/releases and config env based on https://docs.ros.org/en/foxy/index.html, then downloaded other packages to compiled incrementally.
Currently, I am using ros2foxy-6.1(https://github.com/ros2/ros2/releases/download/release-foxy-20211013/ros2-foxy-20211013-windows-release-amd64.zip).
The precomiled has tf2_geometry_msgs, so I don't has this problem, but I also has other problems from new packages when I compile them incrementally.
Compared with Ubuntu, ros2 and relative other thirty-part packages is not friendly enough, so
(1) One package is lost, I download it to compile together.
(2) Small error happens, I modify the code slightly and pull request.

File: ros2/src/airsim_ros_pkgs/src/airsim_ros_wrapper.cpp [hunk 1]
```diff
@@ -908,10 +908,10 @@ rclcpp::Time AirsimROSWrapper::chrono_timestamp_to_ros(const std::chrono::system
 rclcpp::Time AirsimROSWrapper::airsim_timestamp_to_ros(const msr::airlib::TTimePoint& stamp) const
 {
     // airsim appears to use chrono::system_clock with nanosecond precision
-    std::chrono::nanoseconds dur(stamp);
-    std::chrono::time_point<std::chrono::system_clock> tp(dur);
-    rclcpp::Time cur_time = chrono_timestamp_to_ros(tp);
-    return cur_time;
+    //std::chrono::nanoseconds dur(stamp);
+    //std::chrono::time_point<std::chrono::system_clock> tp(dur);
+    //rclcpp::Time cur_time = chrono_timestamp_to_ros(tp);
```
Comment (rajat2004) [inline ros2/src/airsim_ros_pkgs/src/airsim_ros_wrapper.cpp:913]: Better to delete instead of commenting out

**Review by rajat2004 — COMMENTED** [review]
Also tagging @alonfaraj since he did the initial implementation. We should also add atleast Linux ROS2 compilation to Actions

File: ros2/src/airsim_ros_pkgs/src/airsim_ros_wrapper.cpp (hunk 1 above)
Comment (dzywater) [reply ros2/src/airsim_ros_pkgs/src/airsim_ros_wrapper.cpp:913]: I saw all source codes. I found that several calls seem redundant.
Like tf2::Matrix3x3(get_tf2_quat(airlib_quat)).getRPY(roll, pitch, yaw) in AirsimROSWrapper::get_airlib_body_vel_cmd
if only yaw is needed, can use  msr::airlib::VectorMathf::getYaw(airlib_quat) instead.

**Comment by alonfaraj** [comment]
> Also tagging @alonfaraj since he did the initial implementation. We should also add atleast Linux ROS2 compilation to Actions

Done - https://github.com/microsoft/AirSim/pull/4446

**Pull request merged**